### DISCUSSION TITLE
DataSource implementation

### DISCUSSION THREAD

**Discussion by jaclynt18** [awslabs/aws-postgresql-jdbc#33] [Q&A] [answered]
Hi, is there a DataSource implementation that goes along with the new JDBC driver?

**Comment by jaclynt18** [comment, marked as answer]
Nevermind, I think I found the answer.

**Reply by hsuamz** [reply]
Hi @jaclynt18,

Thanks for reaching out and inquiring about the driver. It sounds like you may have found the answer, but we wanted to check in to be sure that you didn't have any other questions regarding your original question or anything else.

To clarify, the driver is intended to be drop-in compatible with the community PostgreSQL driver. It accepts both the original JDBC connection URL and has its own distinct AWS JDBC connection URL. So, it's recognized as a separate datasource to the driver manager.

Since the AWS driver is based on the community PostgreSQL JDBC driver, there can be situations where someone is running both drivers in the same classpath. Please refer to the README.md and look at the connection URL property of `acceptAwsProtocolOnly`.

Please let us know if you have any other questions.

Thanks,

Matthew

**Reply by jaclynt18** [reply]
Hi @hsuamz , 

Thanks for your reply! Just to clarify, should it be possible to use the DataSource-based driver configuration with this new driver? It is my understanding that if this is possible, there should be a DataSource class specific to the AWS JDBC driver, that is different from the current Postgresql DataSource class. Here is an example that shows other popular JDBC drivers and their corresponding DataSource class names: https://github.com/brettwooldridge/HikariCP/blob/dev/README.md#popular-datasource-class-names.

**Reply by hsuamz** [reply]
Hello @jaclynt18,

Thanks for clarifying. Currently, the pre-existing PostgreSQL datasource classes will not work with the AWS protocol since `PGSimpleDataSource` parses for the community connection string. As a workaround, you'd need to specify the AWS JDBC URL to get the enhanced Aurora PostgreSQL failover to be enabled. If Hikari is your use case here, declaring a new `HikariConfig` or `HikariDataSource` and setting the JDBC URL ought to work.

We've been having some discussions with the greater PostgreSQL community and have some changes planned that we will need to make while in preview. I will take note and chat with the team, but having an explicit AWS DataSource can definitely be added.

Let me know if you have further questions.

Thanks,

Matthew

**Reply by jaclynt18** [reply]
Ok, that makes sense, thank you for your quick response!
